### DISCUSSION TITLE
虚拟滚动加载，表头筛选之后无法查询下一页

### DISCUSSION THREAD

**Discussion by skzkk** [dotnetcore/BootstrapBlazor#4282] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

设定 ShowFilterHeader="true" OnQueryAsync="OnQueryAsync" ScrollMode="ScrollMode.Virtual" 之后，不输入任何表头查询条件，正常滚动分页查询，输入条件之后，我重新查询了一次后台数据做筛选，查询到55笔数据之后使用Skip+Take+分页：
1，第一次展示了40笔，鼠标往下滑，不触发第二次分页，导致剩下的数据加载不出来了
2，上下滚动页面，触发了第二次 OnQueryAsync ，分页数据变成了39条（很奇怪，每次数据不一样）

3，设置 Items 参数组件自动实现虚拟滚动的时候，表头筛选没效果，是需要在哪边触发一次筛选呢？

可以用源码复现的，QQ群有发组件源码，也压缩了一下在附件了
[TablesVirtualization.zip](https://github.com/user-attachments/files/16962840/TablesVirtualization.zip)
在第二个demo里，筛选“学历”，每次筛选得到的数据不一样，但是总的数据源Items在页面加载完成就定了，不是很理解

### Expected Behavior

期望：业务是要求一页展示五万笔数据不卡顿，所以目前想到的是滚动加载这个组件，希望是在前端做假分页，页面加载完成把所有数据查出来，但是设置 Items 参数的方式我没能实现表头数据筛选（ShowFilterHeader="true"），更希望能得到这个方案的筛选建议，谢谢

**Comment by bb-auto** [comment]
@skzkk Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by ArgoZhang** [comment]
@skzkk 横向滚动 `Footer` 少了样式。这个是 bug 我另外开 Issue 修复。请提供操作视频。

**Reply by skzkk** [reply]
https://github.com/user-attachments/assets/ff697945-72f3-4611-b9b3-31a8c8208e21
